### ISSUE TITLE
3.1 alpha breaks animations they won't loop properly it holds the last frame after dropping input

### ISSUE THREAD

**Issue by p3terparker** [godotengine/godot#21593]
**Godot version:**
3.1 alpha

**OS/device including version:**
Windows 10 x64

**Issue description:**
Animation doesn't keep looping after you drop input once. 

**Steps to reproduce:**
```
func set_anim(animation):
	if animation != $anim.current_animation:
		$anim.play(animation)

func animate():

	var moving = press_left or press_right #just using the Input.. here
	if moving and is_on_floor():
		set_anim("walk")
	if !moving and is_on_floor():
		set_anim("idle")
```

**Comment by p3terparker** [comment]
might be related to is_on_floor() behavior. I'm getting similar results by changing the safe margins on kinematic body

**Comment by Sslaxx** [comment]
Related to #16250 maybe?

**Comment by DualMatrix** [comment]
Yeah this doesn't seem to be an issue with AnimationPlayer, the problem seems to be with the is_on_floor() function. Do you have a reproduction project?

**Comment by Sslaxx** [comment]
I think a problem with some of these issues is, because it can take a while before they can be looked at, the original poster gives up and moves on. Other than getting more eyes on them, and more dev manpower, dunno what can be done about that; can't always expect patience from the original poster, unfortunately.

**Comment by theblockhouse** [comment]
appeared to be animation related at first, but duplicate of #16250. Changing Safe Margin doesn't seem to help either.
